### DISCUSSION TITLE
How to get the Exception from discord.on_error() ?

### DISCUSSION THREAD

**Discussion by rullinoiz** [Rapptz/discord.py#9458] [Q&A] [answered]
I'm trying to implement custom errors in checks by having raises inside of them like so:
```python
# in modules/checks.py
from discord.ext.commands import Context, check

def is_developer():
    def predicate(ctx: Context):
        if not ctx.author.id == (censored):
            raise AccessDenied(ctx.invoked_with)
        return True
    return check(predicate)
```
```python
# in modules/exceptions.py
class AccessDenied(Exception):
    def __init__(self, action:str=None) -> None:
        self.action = action
        super().__init__('You do not have sufficient permissions to {0}'.format('perform this action.' if not action else f'"{action}".'))
```
however the error is caught by `on_error` rather than `on_command_error`. `on_error` has basically no context and I can't find any way to actually get the Exception from it to then send to the user

(the file in which i want error handling to happen is my main.py file, and these are also prefix-based commands (not app based))

**Comment by Rapptz** [comment, marked as answer]
You use [`sys.exc_info`](https://docs.python.org/3/library/sys.html#sys.exc_info) as [documented](https://discordpy.readthedocs.io/en/stable/api.html#discord.on_error):

![image](https://github.com/Rapptz/discord.py/assets/1695103/88c3d3a1-b286-41fd-a4f1-45dd85a28ecb)

For your custom exception you should derive `CommandError` instead so it goes to the appropriate events, as [documented here](https://discordpy.readthedocs.io/en/stable/ext/commands/commands.html#checks).

**Reply by rullinoiz** [reply]
i started looking differently and reread CommandError and it has exactly what i need, its amazing how you look so hard for a solution but find it right after you ask for help. thanks though, if i hadn't found it then this would've led me in the right direction
